### PR TITLE
Iter15

### PR DESCRIPTION
**Pull request by justEngineer**

### CODE REVIEW

File: internal/database/migrations/0001_init_database.up.sql [hunk 1]
```diff
@@ -0,0 +1,8 @@
+CREATE TABLE IF NOT EXISTS gauge_metrics (
+		id      VARCHAR (255) PRIMARY KEY,
+		value   DOUBLE PRECISION NOT NULL
+	);
+	CREATE TABLE IF NOT EXISTS counter_metrics (
```
Comment (arttet) [inline internal/database/migrations/0001_init_database.up.sql:5]:
```suggestion
CREATE TABLE counter_metrics (
```

File: internal/database/migrations/0001_init_database.up.sql [hunk 2]
```diff
@@ -0,0 +1,8 @@
+CREATE TABLE IF NOT EXISTS gauge_metrics (
```
Comment (arttet) [inline internal/database/migrations/0001_init_database.up.sql:1]:
```suggestion
CREATE TABLE gauge_metrics (
```

File: internal/http/server/handlers/server_handler.go [hunk 3]
```diff
@@ -230,3 +267,44 @@ func (h *Handler) UpdateMetricFromJSON(w http.ResponseWriter, r *http.Request) {
...
+}
+
+func (h *Handler) UpdateMetricsFromBatch(w http.ResponseWriter, r *http.Request) {
+	ctx, cancel := context.WithTimeout(r.Context(), time.Second)
+	defer cancel()
+
+	var metrics []*models.Metrics
+	err := json.NewDecoder(r.Body).Decode(&metrics)
+	if err != nil {
+		h.appLogger.Log.Warn("Error parsing request body as JSON", zap.Error(err))
```
Comment (arttet) [inline internal/http/server/handlers/server_handler.go:287]: It should be the error level.

File: internal/http/server/handlers/server_handler.go [hunk 4]
```diff
@@ -230,3 +267,44 @@ func (h *Handler) UpdateMetricFromJSON(w http.ResponseWriter, r *http.Request) {
...
+	err := h.DBConnection.Ping()
+	if err == nil {
+		w.WriteHeader(http.StatusOK)
+	} else {
+		w.WriteHeader(http.StatusInternalServerError)
+	}
+}
+
+func (h *Handler) UpdateMetricsFromBatch(w http.ResponseWriter, r *http.Request) {
+	ctx, cancel := context.WithTimeout(r.Context(), time.Second)
```
Comment (arttet) [inline internal/http/server/handlers/server_handler.go:281]: It should be in the middleware.

**Review by arttet — APPROVED** [review]

File: internal/http/server/handlers/server_handler.go (hunk 3 above)
Comment (justEngineer) [reply internal/http/server/handlers/server_handler.go]: done

File: internal/http/server/handlers/server_handler.go (hunk 4 above)
Comment (justEngineer) [reply internal/http/server/handlers/server_handler.go]: done

File: internal/database/migrations/0001_init_database.up.sql (hunk 2 above)
Comment (justEngineer) [reply internal/database/migrations/0001_init_database.up.sql]: done

File: internal/database/migrations/0001_init_database.up.sql (hunk 1 above)
Comment (justEngineer) [reply internal/database/migrations/0001_init_database.up.sql]: done

**Pull request merged**